LIPs/lip-6.md

```diff
@@ -0,0 +1,39 @@
+---
+title: Profile Guardian Tied to ERC 6551 Wallet
+description: This LIP proposes to tie the Profile Guardian to an Ethereum Request for Comment (ERC 6551 wallet allowing users to disable it for individual profiles without affecting others
+author: DanIsNearby
+status: Draft
+type: Protocol
+category: Contracts
+created: 2023-11-08
+---
+
+## Abstract
+
+This LIP suggests a modification to the Lens protocol to associate the Profile Guardian with an ERC 6551 wallet. This change will enable users to disable the Profile Guardian for specific profiles, reducing the risk to individual profiles when needed.
+
+## Motivation
+
+The current implementation of the Profile Guardian in Lens ties it to a user's primary wallet, making it an all-or-nothing security measure. This can be problematic when users wish to disable the guardian for a specific profile temporarily. This LIP aims to address this limitation and enhance the flexibility of the Profile Guardian by associating it with an ERC 6551 wallet.
+
+## Specification
+
+The proposed modification involves linking the Profile Guardian to an ERC 6551 wallet. This association will allow users to manage the guardian on a per-profile basis, providing more granular control over their security settings.

```

Comment:
Can you please expand (even in reply to this comment for now) some practical examples of how this might look like?

Just by reading this I would assume you are hinting at a bespoke ERC 6551 the is somewhat Lens Profile aware.

Reply:
From what i understand the Lens profile now is a ERC 6551 and can be used as a wallet store followers store NFTS etc same way as ORB stores their community nfts inside i was thinking of linking the guardian to that entity rather to the wallet holding the profile

Reply:
We were considering having ProfileGuardian as a whole modular system, that will allow different levels of protection:
* Protecting Wallet as a whole (as it is now)
* Protecting Collection as a whole
* Protecting Individual NFT in a Collection

With the latter overriding the former.

I think this LIP might be a special particular case of this system above. I think we don't need to particularly focus on 6551, as focusing on protecting an individual tokenID is basically the same thing, even if it is not 6551 (please correct me if I'm wrong)

Reply:
The 6551 was just an example of how it can be taken care of basically i mean that it would be to unprotect on of the acounts not all of them